lib/src/rules/depend_on_referenced_packages.dart

```diff
@@ -64,26 +68,32 @@ class DependOnReferencedPackages extends LintRule {
   @override
   void registerNodeProcessors(
       NodeLintRegistry registry, LinterContext context) {
-    // Only lint if we have a pubspec.
     var package = context.package;
-    if (package is! PubWorkspacePackage) return;
-    var pubspec = package.pubspec;
-    if (pubspec == null) return;
-    var name = pubspec.name?.value.text;
-    if (name == null) return;
-
-    var dependencies = pubspec.dependencies;
-    var devDependencies = pubspec.devDependencies;
-    var availableDeps = [
-      name,
-      if (dependencies != null)
-        for (var dep in dependencies)
-          if (dep.name?.text != null) dep.name!.text!,
-      if (devDependencies != null &&
-          !isInPublicDir(context.currentUnit.unit, context.package))
-        for (var dep in devDependencies)
-          if (dep.name?.text != null) dep.name!.text!,
-    ];
+    var packageHash = package.hashCode;
```

Comment:
`hashCode` seems like a questionable choice. AFAICT the `package` is inheriting the implementation from `Object`. I don't remember what the lifetime of a `WorkspacePackage` is, but it means that we might get a different hash code for the same set of dependencies (which is inefficient, but no worse than today) and might be able to get the same hash code for a different set of dependencies (which would be breaking).

If you're certain that can't happen (and that no-one will override `hashCode` in those classes in the future), then I guess it's OK. Otherwise, consider implementing a dedicated method.

Reply:
Right.  I was just second-guessing this actually.  An even worse case would be getting the *same* `WorkspacePackage` after it's been changed and so an inconsistent cache.  I'll take a closer look.

Thanks!